app/Http/Controllers/PostController.php

```diff
@@ -21,42 +21,70 @@ class PostController extends Controller
+    protected $answerRepository;
+    protected $tagRepository;
+    protected $userRepository;
+    protected $report;
 
-    public function __construct(PostRepositoryInterface $postRepository)
+    public function __construct(PostRepositoryInterface $postRepository,AnswerRepositoryInterface $answerRepository,
+        TagRepositoryInterface $tagRepository, UserRepositoryInterface $userRepository, Report $report)
     {
         $this->postRepository = $postRepository;
+        $this->answerRepository = $answerRepository;
+        $this->tagRepository = $tagRepository;
+        $this->userRepository = $userRepository;
+        $this->report = $report;
     }
 
     public function index(Request $request)
     {
         $input = $request->all();
         $allPosts = $this->postRepository->getPostWithUserLimit($input);
-        $treadingPost = $this->postRepository->getTreadingPostLimit();
+        $trendingPost = $this->postRepository->gettrendingPostLimit(10);
         $ownTags = '';
 
         if (Auth::check()) {
             $ownTags = $this->postRepository->getOwnTags();
         }
 
-        return view('home.post.index', compact('allPosts', 'input', 'treadingPost', 'ownTags'));
+        return view('home.post.index', compact('allPosts', 'input', 'trendingPost', 'ownTags'));
     }
 
     public function all(Request $request)
     {
         $input = $request->all();
         $allPosts = $this->postRepository->getPostWithUserPaginate($input);
-        $treadingPost = $this->postRepository->getTreadingPostLimit();
+        $trendingPost = $this->postRepository->gettrendingPostLimit(10);
         $ownTags = '';
 
         if (Auth::check()) {
             $ownTags = $this->postRepository->getOwnTags();
         }
 
-        return view('home.post.all', compact('allPosts', 'input', 'treadingPost', 'ownTags'));
+        return view('home.post.all', compact('allPosts', 'input', 'trendingPost', 'ownTags'));
     }
 
-    public function detail()
+    public function detail($id)
     {
-        return view('home.post.detail');
+        $this->postRepository->increaseViewTotal($id);
+        $post = $this->postRepository->findPostWithUser($id);
+
+        if (empty($post)) {
+            return redirect()->route('home.post.index');
+        }
+        $postReplies = $this->postRepository->getRepliesFromPost($post);
+        $postVote = $this->postRepository->findVoteFromPost($post);
+        //get answer from post id
+        $answers = $this->answerRepository->getAnswerFromPost($id);
+        $answersReplies = $this->answerRepository->getRepliesFromAnswers($answers);
+        $answersVotes = $this->answerRepository->getVotesFromAnswers($answers);
+        $relatedPost = $this->postRepository->getRelatedPost($post->tags, $id);
+
+        if (!empty($post)) {
+            return view('home.post.detail', compact('post', 'postReplies', 'postVote',
+                'answers', 'answersReplies', 'answersVotes', 'relatedPost'));
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 85</br>Multi-line function call not indented correctly; expected 12 spaces but found 16